tests/fail/tree_borrows/raw_write_invalidates_read.rs

```diff
@@ -0,0 +1,12 @@
+//@compile-flags: -Zmiri-tree-borrows
+
+// Write to raw pointer invalidates all shared references derived from it.
+fn main() {
+    let mut root = 6u8;
+    let mref = &mut root;
+    let ptr = mref as *mut u8;
+
+    let shr1 = unsafe { &*ptr };
+    unsafe { *ptr = 2; } // Parent Write
+    let _v = *shr1; //~ ERROR: /read access through .* is forbidden/
+}
```

Comment:
I like the test but not the filename. "raw write" isn't the key part... and it's also "parent write invalidates read" but that file already exists. ;)

What about parent_write_invalidates_mut and parent_write_invalidates_shr?

Reply:
In fact isn't this test here a duplicate of shr_frozen_violation2?

Reply:
> `parent_write_invalidates_mut_raw`, `parent_write_invalidates_shr`

Not sure about those names. It's not a parent write that invalidates a raw mutable pointer, it's a write _to a raw mutable parent_ that invalidates a shr.

It's also not quite the same as `shr_frozen_violation2`:
```
parent_write_invalidates_shr:
    parent                                    <--- (1) write
         |--- shr child
               = raw child                 <---  (2) read
         
raw_write_invalidates_read:
    parent
    = raw parent                      <--- (1) write
         |--- shr child                 <--- (2) read

shr_frozen_violation2:
    parent                                <--- (1) write
    = raw parent
        |--- shr child                 <--- (2) read
```
so on the surface those three tests all check that a parent write invalidates a child read, but also they check that the parent-child link is in the right place in a chain of reference-raw-reference.

I can accept though that this might not be the place to test that and I'm not opposed to deleting `raw_write_invalidates_read` in favor of `shr_frozen_violation2`.

Reply:
Given that raw ptrs inherit their parent tag, those all look exactly identical? Write to parent, then read from frozen child?